The "Mac OS X kludge" is only needed for Mac OS X 10.5 and earlier.
--- Makefile.orig	2011-07-22 19:20:21.000000000 -0500
+++ Makefile	2020-11-21 17:28:02.000000000 -0600
@@ -419,7 +419,7 @@
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
 	PASSWDTYPE=pam \
-	EXTRACFLAGS="$(EXTRACFLAGS) -DMAC_OSX_KLUDGE=1" \
+	EXTRACFLAGS="$(EXTRACFLAGS)" \
 	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib PAMDLFLAGS=-lpam"
 
 osx:	osxok an
